File: src/engine/AlgorithmRegistry.cpp
#include "AlgorithmRegistry.h"
#include "../algorithms/BubbleSort.h"
#include "../algorithms/InsertionSort.h"
#include "../algorithms/SelectionSort.h"
#include "../algorithms/QuickSort.h"
#include "../algorithms/HeapSort.h"
#include "../algorithms/MergeSort.h"
#include "../algorithms/CountingSort.h"


void AlgorithmRegistry::registerSortingAlgorithms() {
    sortingAlgorithms.push_back(std::make_unique<BubbleSort>());
    sortingAlgorithms.push_back(std::make_unique<InsertionSort>());
    sortingAlgorithms.push_back(std::make_unique<SelectionSort>());
    sortingAlgorithms.push_back(std::make_unique<QuickSort>());
    sortingAlgorithms.push_back(std::make_unique<HeapSort>());
    sortingAlgorithms.push_back(std::make_unique<MergeSort>());
    sortingAlgorithms.push_back(std::make_unique<CountingSort>());
}

const std::vector<std::unique_ptr<SortingAlgorithm>>&
AlgorithmRegistry::getSortingAlgorithms() const {
    return sortingAlgorithms;
}

File: java/algovista/engine/AlgorithmRegistry.java
package algovista.engine;

import algovista.algorithms.BubbleSort;
import algovista.algorithms.CountingSort;
import algovista.algorithms.HeapSort;
import algovista.algorithms.InsertionSort;
import algovista.algorithms.MergeSort;
import algovista.algorithms.QuickSort;
import algovista.algorithms.SelectionSort;
import algovista.algorithms.SortingAlgorithm;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class AlgorithmRegistry {
    private final List<SortingAlgorithm> sortingAlgorithms = new ArrayList<>();

    public void registerSortingAlgorithms() {
        sortingAlgorithms.add(new BubbleSort());
        sortingAlgorithms.add(new InsertionSort());
        sortingAlgorithms.add(new SelectionSort());
        sortingAlgorithms.add(new QuickSort());
        sortingAlgorithms.add(new HeapSort());
        sortingAlgorithms.add(new MergeSort());
        sortingAlgorithms.add(new CountingSort());
    }

    public List<SortingAlgorithm> getSortingAlgorithms() {
        return Collections.unmodifiableList(sortingAlgorithms);
    }
}
